vyper/compiler/output.py

```diff
@@ -159,6 +161,19 @@ def build_interface_output(compiler_data: CompilerData) -> str:
     return out
 
 
+def _get_structs(m: ModuleT, visited: set[ModuleT] = None) -> [StructT]:
+    visited = visited or set()
+    if m in visited:
+        return []
+    visited.add(m)
+    structs = list(m.interface.structs.values())
```

Comment:
hmm. what if two modules have a struct with the same name?

Reply:
yeah, that's what i was getting at with https://github.com/vyperlang/vyper/pull/4362#issuecomment-2480532047, I'll prepend it.